modules/networking/virtual_network_gateways/module.tf

```
@@ -45,8 +45,11 @@ resource "azurerm_virtual_network_gateway" "vngw" {
       radius_server_secret  = try(vpn_client_configuration.value.radius_server_secret, null)
 
       root_certificate {
-        name             = vpn_client_configuration.value.root_certificate.name
-        public_cert_data = vpn_client_configuration.value.root_certificate.public_cert_data
+        name = vpn_client_configuration.value.root_certificate.name
+        public_cert_data = try(
```

This code should produce get the value at plan time. Let me know if it is working

```suggestion
        public_cert_data = can(vpn_client_configuration.key) ? replace(replace(data.azurerm_key_vault_secret.vpn_client_configuration_root_certificate[vpn_client_configuration.key].value, "-----BEGIN CERTIFICATE-----", ""), "-----END CERTIFICATE-----", "") : vpn_client_configuration.value.root_certificate.public_cert_data
```